File: CodeChef/June Challenge 2016/frjump(partially solved)/frjump.cpp
#include<iostream>
#include<cmath>
#include<iomanip>
#define MOD 1000000007
using namespace std;
int main()
{
	long long n,q,i,j,k,typ,first,sec;
	cin>>n;
	long long *f=new long long[n];
	for(i=0;i<n;i++)
	{
		cin>>f[i];
	}
	cin>>q;
	while(q--)
	{
		cin>>typ;
		if(typ==1)
		{
			cin>>first>>sec;
			f[first-1]=sec;
		}
		else if(typ==2)
		{
			cin>>first;
			long long res=1;
			for(i=0;i<n;i+=first)
			{
				res*=f[i];
			}
			long double temp=(long double)(log10(res));
			long double frac=temp-floor(temp);
			long double ans1=(long double)(pow(10,frac+1e-10));
			int ans=floor(ans1);
			
			//cout<<"inter: "<<temp<<" "<<frac<<" "<<ans1<<" "<<ans<<endl;
			res%=MOD;
			cout<<ans<<" "<<res<<endl;
		}
	}
	return 0;
}

File: CodeChef/June Challenge 2016/frjump(partially solved)/frjump.py
from math import *
n=(long)(raw_input())

lis=raw_input()
li=lis.split()
for i in xrange(0,n):
	li[i]=(long)(li[i])
q=(long)(raw_input())
for i in xrange(0,q):
	val=raw_input()
	lisval=val.split()
	for j in xrange(0,len(lisval)):
		lisval[j]=(long)(lisval[j])
	
	if(lisval[0]==1):
		
		li[lisval[1]-1]=lisval[2]
	elif(lisval[0]==2):
		
		res=1
		for j in xrange(0,n,lisval[1]):
			res*=li[j]
		temp=log10(res)
		frac=temp-floor(temp)
		ans1=10**(frac+1e-10)
		ans=(int)(ans1)
		res%=(1000000007)
		print "%d"%(ans),
		
		print "%d"%(res)
		

